internal/incident/restore_incidents_test.go

```diff
@@ -0,0 +1,238 @@
+func makeIncident(ctx context.Context, db *icingadb.DB, t *testing.T, recovered bool) *Incident {
+	ev := &event.Event{
+		Time:     time.Time{},
+		SourceId: 1,
+		Name:     makeRandomString(t),
+		Tags: map[string]string{ // Always generate unique object tags not to produce same sha1!
+			"host":    makeRandomString(t),
+			"service": makeRandomString(t),
+		},
+		ExtraTags: map[string]string{
+			"hostgroup/database-server": "",
+			"servicegroup/webserver":    "",
+		},
+	}
+
+	o, err := object.FromEvent(ctx, db, ev)
+	require.NoError(t, err)
+
+	i := NewIncident(db, o, &config.RuntimeConfig{}, nil)
+	i.StartedAt = types.UnixMilli(time.Unix(time.Now().Add(-2*time.Hour).Unix(), 0))
+	i.Severity = event.SeverityCrit
+	if recovered {
+		i.Severity = event.SeverityOK
+		i.RecoveredAt = types.UnixMilli(time.Unix(time.Now().Unix(), 0))
+	}
+
+	tx, err := db.BeginTxx(ctx, nil)
+	require.NoError(t, err, "starting a transaction should not fail")
+	require.NoError(t, i.Sync(ctx, tx), "failed to insert incident")
+	require.NoError(t, tx.Commit(), "committing a transaction should not fail")
+
+	return i
+}
+
+func makeRandomString(t *testing.T) string {
+	buf := make([]byte, 20)
+	_, err := rand.Read(buf)
+	require.NoError(t, err, "failed to generate random string")
+
+	return fmt.Sprintf("%x", buf)
+}
+
+// getTestDB retrieves the database config from env variables, opens a new database and returns it.
+//
+// The test suite will be skipped if no environment variable is set, otherwise fails fatally when
+// invalid configurations are specified.
+func getTestDB(ctx context.Context, t *testing.T) *icingadb.DB {
+	if _, ok := os.LookupEnv("NOTIFICATIONS_TESTS_DB_TYPE"); !ok {
+		t.Skipf("Environment %q not set, skipping test!", "NOTIFICATIONS_TESTS_DB_TYPE")
+	}
+
+	c := &baseConfig.Database{}
+	if err := defaults.Set(c); err != nil {
+		t.Skipf("Skipping test due to an error: %s", err.Error())
+	}
+
+	dbEnvs := []string{
+		"NOTIFICATIONS_TESTS_DB_TYPE", "NOTIFICATIONS_TESTS_DB", "NOTIFICATIONS_TESTS_DB_USER",
+		"NOTIFICATIONS_TESTS_DB_PASSWORD", "NOTIFICATIONS_TESTS_DB_HOST", "NOTIFICATIONS_TESTS_DB_PORT",
+	}
+
+	for _, env := range dbEnvs {
+		v, ok := os.LookupEnv(env)
+		if !ok {
+			continue
+		}
+
+		switch env {
+		case "NOTIFICATIONS_TESTS_DB_TYPE":
+			c.Type = strings.ToLower(v)
+		case "NOTIFICATIONS_TESTS_DB":
+			c.Database = v
+		case "NOTIFICATIONS_TESTS_DB_USER":
+			c.User = v
+		case "NOTIFICATIONS_TESTS_DB_PASSWORD":
+			c.Password = v
+		case "NOTIFICATIONS_TESTS_DB_HOST":
+			c.Host = v
+		case "NOTIFICATIONS_TESTS_DB_PORT":
+			port, err := strconv.Atoi(v)
+			if err != nil {
+				require.NoError(t, err, "invalid port provided")
+			}
+
+			c.Port = port
+		}
+	}
+
+	if err := c.Validate(); err != nil {
+		require.NoError(t, err, "database config validation should not fail")
+	}
+
+	db, err := c.Open(logging.NewLogger(zap.NewNop().Sugar(), time.Hour))
```

Comment:
`zap` allows to create a logger for tests, have you considered using it? https://pkg.go.dev/go.uber.org/zap/zaptest#NewLogger

Reply:
NVM! It's already there as part of `zap` main package.